gems/smithy-schema/lib/smithy-schema/document/serializer.rb

```diff
@@ -0,0 +1,283 @@
+# frozen_string_literal: true
+
+require 'base64'
+require 'time'
+
+module Smithy
+  module Schema
+    module Document
+      # Serializes data into a document data.
+      class Serializer
+        include Shapes
+
+        # @param [TypeRegistry] type_registry Used to find shape based on
+        #  on document discriminator.
+        def initialize(type_registry)
+          @type_registry = type_registry
+        end
+
+        # Create document data from various input data formats
+        # @param [Object] data Input data can be: Ruby objects, instance of a runtime shape or a
+        #  JSON response with type discriminator.
+        # @return [Data] document data
+        #
+        # @example Ruby objects as input
+        #   # create serializer with an existing type registry
+        #   serializer = Smithy::Schema::Document::Serializer(type_registry)
+        #
+        #   # ruby objects as input
+        #   serializer.create_document("some document")
+        #   # => {"foo" => "bar"}
```

Comment:
This example is not really helpful. How did you get from a string to a hash here?